python/lsst/analysis/tools/actions/plot/coaddDepthPlot.py

```diff
@@ -0,0 +1,159 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("CoaddDepthPlot",)
+
+from typing import TYPE_CHECKING, Any, Mapping
+
+from matplotlib.figure import Figure
+from matplotlib.lines import Line2D
+import matplotlib.pyplot as plt
+
+from lsst.utils.plotting import publication_plots, set_rubin_plotstyle
+
+from ...interfaces import PlotAction, Vector
+from ..vector import BandSelector, PatchSelector
+from .plotUtils import addPlotInfo
+
+if TYPE_CHECKING:
+    from ...interfaces import KeyedData, KeyedDataSchema
+
+bands_dict = publication_plots.get_band_dicts()
+
+
+class CoaddDepthPlot(PlotAction):
+    """Make a plot of pixels per coadd depth."""
+
+    def setDefaults(self):
+        super().setDefaults()
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        base: list[tuple[str, type[Vector]]] = []
+        base.append(("patch", Vector))
+        base.append(("band", Vector))
+        base.append(("depth", Vector))
+        base.append(("pixels", Vector))
+        return base
+
+    def __call__(self, data: KeyedData, **kwargs) -> Figure:
+        self._validateInput(data)
+        return self.makePlot(data, **kwargs)
+
+    def _validateInput(self, data: KeyedData) -> None:
+        needed = set(k[0] for k in self.getInputSchema())
+        if not needed.issubset(data.keys()):
+            raise ValueError(f"Input data does not contain all required keys: {self.getInputSchema()}")
+
+    def makePlot(self, data: KeyedData, plotInfo: Mapping[str, str] | None = None, **kwargs: Any) -> Figure:
+        """Make the plot.
+
+        Parameters
+        ----------
+        `KeyedData`
+            The catalog to plot the points from.
+
+        plotInfo : `dict`
+            A dictionary of the plot information.
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            The resulting figure.
+        """
+        set_rubin_plotstyle()
+        fig = plt.figure(dpi=300, figsize=(20, 20))
+
+        max_depth = max(data['depth'])
+        max_pixels = max(data['pixels'])
+
+        plt.subplots_adjust(hspace=0, wspace=0)
+
+        patch_counter = 90  # The top left corner of a tract is patch 90
+        m = 0  # subplot index
+        while patch_counter >= 0:
+            for n in range(10):  # column index
+                ax = plt.subplot(10, 10, m + 1)  # there are 10x10 patches per tract
+                patchSelector = PatchSelector(vectorKey='patch', patches=[patch_counter])
```

Comment:
There's no point making a `ConfigurableAction` inside a call method where it can't actually be reconfigured in a pipeline. You already put `patch` and `band` in the schema so you may as well just do `patch_mask = data["patch"] == patch_counter`, and same for `band_mask`.